torchdata/dataloader2/reading_service.py

```diff
@@ -187,10 +204,13 @@ def initialize(self, datapipe: DataPipe) -> DataPipe:
             )
             self.datapipes.append(local_datapipe)
 
-        return _IterateQueueDataPipes(self.datapipes)  # type: ignore[return-value]
+        self.datapipes_iterator = _IterateQueueDataPipes(self.datapipes)
```

Comment:
nit: Could we please rename it as it's still a `DataPipe` not an `Iterator`?

Reply:
Done